### PR TITLE
xorg: Remove xcb-proto dependency

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

File: xcb-proto.rb [hunk 1]
```diff
@@ -3,35 +3,15 @@ class XcbProto < Formula
...
-
-  patch :p1 do
-    url "http://www.linuxfromscratch.org/patches/blfs/svn/xcb-proto-1.12-python3-1.patch"
-    sha256 "4eef0285392c525fcb7aeb54c1b6dc8406dc33c5845c1311b65b3efc409bea2e"
-  end
-
-  patch :p1 do
-    url "http://www.linuxfromscratch.org/patches/blfs/svn/xcb-proto-1.12-schema-1.patch"
-    sha256 "3bc1a1871d17325d1f591a8ec2091f956b2071d83a6e9998d7d4880c4abf3e8b"
-  end
```
Comment (maxim-belkin) [inline xcb-proto.rb]: I think this patch is still required

Comment (maxim-belkin) [reply xcb-proto.rb]: Oh, well... it is needed for tests, which are deleted below, so... disregard the above comment :)

Comment (maxim-belkin) [reply xcb-proto.rb]: let's maybe delete everything related to python, but leave libxml and tests

**Review by maxim-belkin — APPROVED** [review]

File: xcb-proto.rb [hunk 2]
```diff
@@ -3,35 +3,14 @@ class XcbProto < Formula
   homepage "https://www.x.org/"
   url "https://xcb.freedesktop.org/dist/xcb-proto-1.12.tar.bz2"
   sha256 "5922aba4c664ab7899a29d92ea91a87aa4c1fc7eb5ee550325c3216c480a4906"
-  revision 1
+  revision 2
   # tag "linuxbrew"
 
   bottle do
     cellar :any_skip_relocation
-    sha256 "1b2b12848e39c48870a3db864572e0f4eb63fd4f9d7c715fe7765b302c378c33" => :x86_64_linux
```
Comment (sjackman) [inline xcb-proto.rb]: No need to remove the bottle.

File: xorg.rb [hunk 3]
```diff
@@ -63,6 +56,7 @@ class Xorg < Formula
   depends_on "linuxbrew/xorg/libpciaccess" => args_check_static
   depends_on "linuxbrew/xorg/libxkbfile" => args_check_static
   depends_on "linuxbrew/xorg/libxshmfence" => args_check_static
+  depends_on "linuxbrew/xorg/xcb-proto" => args_check_static
```
Comment (sjackman) [inline xorg.rb]: I'm fine with adding this line, but I'm curious, did you find a formula that needs it?

File: xcb-proto.rb (hunk 1 above)
Comment (maxim-belkin) [reply xcb-proto.rb]: So, we do need python. Its version does not matter but if we've got python3, then we need to apply a patch for python3.

**Comment by sjackman** [comment]
I'm happy with these changes once CI is 🍏

**Comment by sjackman** [comment]
CircleCI
```
==> Installing xcb-proto from linuxbrew/xorg
git config --replace-all homebrew.private false
==> Downloading https://xcb.freedesktop.org/dist/xcb-proto-1.12.tar.bz2
Already downloaded: /home/linuxbrew/.cache/Homebrew/xcb-proto-1.12.tar.bz2
==> Verifying xcb-proto-1.12.tar.bz2 checksum
tar xjf /home/linuxbrew/.cache/Homebrew/xcb-proto-1.12.tar.bz2
==> ./configure --prefix=/home/linuxbrew/.linuxbrew/Cellar/xcb-proto/1.12_2 --sysconfdir=/home/linuxbrew/.linuxbrew/etc --localstatedir=/home/linuxbrew/.linuxbrew/var --disable-silent-rules
.
Last 150 lines from /home/linuxbrew/.cache/Homebrew/Logs/xcb-proto/01.configure:
2018-02-05 21:41:46 +0000

./configure
--prefix=/home/linuxbrew/.linuxbrew/Cellar/xcb-proto/1.12_2
--sysconfdir=/home/linuxbrew/.linuxbrew/etc
--localstatedir=/home/linuxbrew/.linuxbrew/var
--disable-silent-rules

checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for xmllint... no
configure: WARNING: xmllint not found; unable to validate against schema.
checking for a Python interpreter with version >= 2.5... none
configure: error: no suitable Python interpreter found
```
https://circleci.com/gh/Linuxbrew/homebrew-xorg/963
https://circleci.com/api/v1.1/project/github/Linuxbrew/homebrew-xorg/963/output/111/0?file=true

**Comment by sjackman** [comment]
TravisCI
```
/usr/lib/ruby/1.9.1/rubygems/version.rb:191:in `strip!': can't modify frozen String (RuntimeError)
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! You're running 1.9.3. (RuntimeError)
```
https://travis-ci.org/Linuxbrew/homebrew-xorg/builds/337755276#L1550

This bug is fixed upstream in Homebrew/brew. It's easily worked around here though.
See https://github.com/brewsci/homebrew-bio/blob/master/.travis.yml
```yaml
rvm: system
…
  - export HOMEBREW_FORCE_VENDOR_RUBY=1
```

**Comment by iMichka** [comment]
@sjackman @maxim-belkin are you OK with the new version of this?

**Review by sjackman — APPROVED** [review]

**Comment by sjackman** [comment]
Looks good to me. Thanks, Michka.
Do either you or Maxim have the time/interest to the the TravisCI failure? It should be a two-line fix.
See https://github.com/Linuxbrew/homebrew-xorg/pull/348#issuecomment-363249607

**Comment by iMichka** [comment]
It would also maybe cool if we could move the formulae to a Formula folder, and bring the travis/circle files in sync with core. Not sure I have much time at the moment though.

**Comment by sjackman** [comment]
Agreed, moving formulae to `Formula/` would be great. Maxim, you interested?

**Pull request closed without merging**